app/src/main/java/com/kabouzeid/gramophone/ui/activities/AlbumDetailActivity.java

```diff
@@ -268,11 +268,11 @@ public void onResponse(Call<LastFmAlbum> call, Response<LastFmAlbum> response) {
                         wikiReady();
                         return;
                     }
-                    else if(call.request().url().queryParameter("lang") != null) {
-                        //If the "lang" parameter is set and no wiki is given, retry with default language
-                        lastFMRestClient.getApiService().getAlbumInfo(getAlbum().getTitle(), getAlbum().getArtistName(), null).enqueue(this);
-                        return;
-                    }
+                }
+                if(call.request().url().queryParameter("lang") != null) {
```

Comment:
@Materight I pushed some minor refactoring/code cleanup changes to this branch. Let me know if the modified code (https://github.com/kabouzeid/Phonograph/pull/178/commits/0bd8e4292d1429f96ef1ffb00475dc21f4dc36cc#diff-6fa2840379ca15958a5b65fb6e3dc6c0R276) still makes sense for this.